ext/standard/array.c

```diff
@@ -2057,13 +2055,25 @@ PHP_FUNCTION(array_fill_keys)
 }
 /* }}} */
 
-#define RANGE_CHECK_INIT_ARRAY(start, end) do { \
+#define RANGE_CHECK_DOUBLE_INIT_ARRAY(start, end) do { \
 		double __calc_size = ((start - end) / step) + 1; \
-		if (fabs(__calc_size) >= (double)HT_MAX_SIZE) { \
-			php_error_docref(NULL, E_WARNING, "The supplied range exceeds the maximum array size: start=%0.0f end=%0.0f", start > end ? end : start, start > end ? start : end); \
+		if (__calc_size > (double)HT_MAX_SIZE) { \
```

Comment:
Ok, I've corrected the condition to ensure  `__calc_size` is less than `HT_MAX_SIZE` with 55f6132.

Thanks.